code/script.js

```diff
@@ -229,97 +237,171 @@ const setSecret = () => {
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  board.style.display = 'flex'
+  winOrLose.style.display = 'none'
+  charactersInPlay = CHARACTERS // Here we're setting charactersInPlay array to be all the characters to start with
+  generateBoard()
+  setSecret() 
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label // This variable stores what option group (category) the question belongs to.
+  const value = questions.value // We also need a variable that stores the actual value of the question we've selected.
 
-  currentQuestion = {
+  currentQuestion = { // currentQuestion object created, before it was just a variable that we declared
     category: category,
-    // value: value
+    value: value,
   }
+  console.log(currentQuestion)
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+    if (category === 'hair' || category === 'eyes') {
+      if (secret[category] === value) {
+        filterCharacters(true)
+      } 
+      else {
+        filterCharacters(false)
+      }
+       
+    } else if (category === 'accessories' || category === 'other') {
+        if (secret[category].includes(value)) {
+          filterCharacters(true)
+        }
+        else {
+          filterCharacters(false)
+        }
+      } 
+}        
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+  //Alternative way for checkQuestion
+  //  let keep = false
+    
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  //   if (category === 'hair' || category === 'eyes') {
+  //     keep = (value === secret[category])
+  //     console.log(keep)
 
-  } else if (category === 'accessories' || category === 'other') {
+  //   } else if (category === 'accessories' || category === 'other') {
+  //     keep = secret[category].includes(value)
+  //   }
```

Comment:
These if statements are not commented and I saw that you have a code above that is working. I wonder if you just forgot to comment these lines out? :)